components/kserve/kserve.go

```diff
@@ -198,3 +187,24 @@ func (k *Kserve) removeServerlessFeatures(instance *dsciv1.DSCInitializationSpec
 	}
 	return nil
 }
+
+func checkDepedentOps(cli client.Client) *multierror.Error {
+	var multiErr *multierror.Error
+
+	if found, err := deploy.OperatorExists(cli, ServiceMeshOperator); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	} else if !found {
+		err = fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
+			ServiceMeshOperator, ComponentName)
+		multiErr = multierror.Append(multiErr, err)
+	}
+
+	if found, err := deploy.OperatorExists(cli, ServerlessOperator); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	} else if !found {
+		err = fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
+			ServerlessOperator, ComponentName)
+		multiErr = multierror.Append(multiErr, err)
+	}
+	return multiErr
+}
```

Comment:
I think returning an error is enough here. I added a bit extra :)

```suggestion
func checkRequiredOperatorsInstalled(cli client.Client) error {
	 var multiErr *multierror.Error
   
	checkAndAppendError := func(operatorName string) {
	    if found, err := deploy.OperatorExists(cli, operatorName); err != nil {
	        multiErr = multierror.Append(multiErr, err)
	    } else if !found {
	        err = fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
	            operatorName, ComponentName)
	        multiErr = multierror.Append(multiErr, err)
	    }
	}
	
	checkAndAppendError(ServiceMeshOperator)
	checkAndAppendError(ServerlessOperator)
	
	return multiErr.ErrorOrNil()
}
```